peps/pep-0748.rst

```diff
@@ -0,0 +1,923 @@
+as described in the “Runtime” section below.
+
+Specifically, this class does not parse any provided input to validate that it
+is a correct certificate, and also does not provide any form of introspection
+into a particular certificate. Backends are not required to provide such
+introspection either. Peer certificates that are received during the handshake
+are provided as raw DER bytes.
+
+Future versions of the API may provide alternative constructors, e.g. to load
+certificates from HSMs, if a common interface emerges for doing this.
+
+.. code-block:: python
+
+    TODO Certificate from tlslib
+
+Private Keys
+~~~~~~~~~~~~
+
+This module would define a concrete private key class. Much like the
+``Certificate`` class, this class has three attributes to correspond to the
+three constructors, and further has all the caveats of the ``Certificate``
+class.
+
+.. code-block:: python
+
+    TODO PrivateKey from tlslib
+
+Signing Chain
+~~~~~~~~~~~~~
+
+In order to authenticate themselves, TLS participants need to provide a leaf
+certificate with a chain leading up to some root certificate that is trusted by
+the other side. Servers always need to authenticate themselves to clients, but
+clients can also authenticate themselves to servers during client
+authentication. Additionally, the leaf certificate must be accompanied by a
+private key, which can either be stored in a separate object, or together with
+the leaf certificate itself. This module defines the collection of these objects
+as a ``SigningChain`` as detailed below:
+
+.. code-block:: python
+
+    TODO SigningChain from tlslib
+
+As shown in the configuration classes above, a client can have one signing chain
+in the case of client authentication or none otherwise. A server can have a
+sequence of signing chains, which is useful when it is responsible for multiple
+domains.
+
+Trust Store
+~~~~~~~~~~~
+
+As discussed above, loading a trust store represents an issue because different
+TLS implementations vary wildly in how they allow users to select trust stores.
+For this reason, we need to provide a model that assumes very little about the
+form that trust stores take.
+
+This problem is the same as the one that the ``Certificate`` and ``PrivateKey``
+types need to solve. For this reason, we use the exact same model, by creating a
+concrete class that captures the various means of how users could define a trust
+store.
+
+A given TLS implementation is not required to handle all possible trust stores.
+However, it is strongly recommended that a given TLS implementation handles the
+``system`` constructor if at all possible, as this is the most common validation
+trust store that is used. Backends can communicate unsupported options as
+described in the “Runtime” section below.
+
+.. code-block:: python
+
+    TODO TrustStore from tlslib
+
+Runtime Access
+~~~~~~~~~~~~~~
+
+A not-uncommon use case for library users is to want to allow the library to
```

Comment:
```suggestion
A not-uncommon use case is for library users to want to specify the TLS backend to use while allowing the library to configure the details of the actual TLS connection. 
```

(no multiline comments in the mobile web view, this suggestion is to replace the entire first sentence)